staging/src/k8s.io/legacy-cloud-providers/azure/azure_utils.go

```diff
@@ -74,3 +77,21 @@ func (lm *lockMap) unlockEntry(entry string) {
 func getContextWithCancel() (context.Context, context.CancelFunc) {
 	return context.WithCancel(context.Background())
 }
+
+// aggregateGoroutinesWithDelay aggregates goroutines and runs them
+// in parallel with delay before starting each goroutine
+func aggregateGoroutinesWithDelay(delay time.Duration, funcs ...func() error) utilerrors.Aggregate {
```

Comment:
could you add a uni test for the new func?

Reply:
added the same unit test the other func was using. Will see if we can test anything more in this func.